transformations/style_paraphraser/transformation.py

```diff
@@ -0,0 +1,251 @@
+}
+
+
+class StyleTransferParaphraser(SentenceOperation):
+    tasks = [TaskType.TEXT_CLASSIFICATION, TaskType.TEXT_TO_TEXT_GENERATION]
+    languages = ["en"]
+    heavy = True
+    keywords = [
+        "model-based",
+        "transformer-based",
+        "tokenizer-required",
+        "unnatural-sounding",
+        "unnaturally-written",
+        "possible-meaning-alteration",
+        "high-coverage",
+    ]
+    """
+        Style transfer paraphraser, using a GPT2-model of choice.
+
+        Args:
+            style : str
+                The style to use. Options include Bible, Shakespeare, Basic, Romantic Poetry, Switchboard and Tweets.
+            device : device to use for computations.
+                Default: None, and it will then resort to CUDA if available, else CPU.
+            upper_length :
+                The maximum length.
+                Options: "eos" or "same_N" (e.g., "same_5"), where N will be the max_length.
+            beam_size : size of the beam during beam search (if top_p == 0.0)
+                Default: 1
+            top_p : float
+                top_p sampling, between 0.0 and 1.0
+                Default: 0.0 (meaning using a greedy approach)
+            temperate : float
+                Sampling temperate
+                Default: 0.0
+    """
+
+    def __init__(
+        self,
+        style: str = "Basic",
+        device=None,
+        upper_length="same_5",
+        beam_size: int = 1,
+        top_p: int = 0.0,
+        temperature: float = 0.0,
+    ):
+        try:
+            nltk.data.find("tokenizers/punkt")
+        except LookupError:
+            nltk.download("punkt")
+        self.style = style
+
+        assert (
+            style in MODELS_SUPPORTED.keys()
+        ), f"Style not supported. The following styles are supported: {', '.join(list(MODELS_SUPPORTED.keys()))}"
+        model_path = MODELS_SUPPORTED[style]
+        self.args = {}
+        self.device = device
+        if self.device is None:
+            self.device = torch.device(
+                "cuda" if torch.cuda.is_available() else "cpu"
+            )
+        self.args["upper_length"] = upper_length
+        self.args["stop_token"] = "eos" if upper_length == "eos" else None
+        self.args["beam_size"] = beam_size
+        self.args["temperature"] = temperature
+        self.args["top_p"] = top_p
+        self.args["top_k"] = 1
+        self.args["device"] = self.device
+        self.config = BASE_CONFIG
+
+        self.config["global_dense_length"] = 0
+        model = GPT2LMHeadModel.from_pretrained(model_path)
+        model.to(self.device)
+        self.gpt2_model = GPT2ParentModule(gpt2=model, device=device)
+        self.tokenizer = GPT2Tokenizer.from_pretrained(model_path)
+
+    def modify_p(self, top_p):
+        """Set top_p to another value"""
+        self.args["top_p"] = top_p
+
+    def generate(self, sentence, top_p=None, n_samples: int = 1):
```

Comment:
Could you please change this `n_samples` to `max_outputs`, so that it would use the same variable name as other transformations?

Reply:
Fixed, and good point!